app/jobs/scheduled/clean_up_uploads.rb

```diff
@@ -71,7 +54,9 @@ def execute(args)
         .where("g.flair_upload_id IS NULL")
         .where("ss.value IS NULL")
 
-      result = result.where("uploads.url NOT IN (?)", ignore_urls) if ignore_urls.present?
+      if SiteSetting.selectable_avatars.present?
+        result = result.where("uploads.id NOT IN (?)", SiteSetting.selectable_avatars.map(&:id))
```

Comment:
You don't need to resort to SQL here. The following should work:

```suggestion
        result = result.where.not(id: SiteSetting.selectable_avatars.map(&:id))
```
```